agent/doc/swdesign/README.md

```diff
@@ -161,6 +170,462 @@ Needs:
+
+Tags:
+- WorkloadScheduler
+
+Needs:
+- impl
+- utest
+- stest
+
+#### The agent shall report workload state stopping for updated workloads with unfulfilled DeleteConditions
+`swdd~agent-reports-pending-delete-workload-state-on-pending-update-delete~1`
+
+Status: approved
+
+When the WorkloadScheduler receives the `WorkloadOperation` to update a workload
+and the workload is an inter-workload dependency of at least one other workload
+and the workload has at least one unfulfilled `DeleteCondition`,
+then the WorkloadScheduler shall report the workload state `Stopping(WaitingToStop)`.
+
+Rationale: This communicates to the user that the delete of the inter-workload dependency is scheduled but not ready because another workload requires it.
+
+Tags:
+- WorkloadScheduler
+
+Needs:
+- impl
+- utest
+- stest
+
+#### The agent shall report workload state pending for updated workloads with fulfilled DeleteConditions
+`swdd~agent-reports-pending-create-workload-state-on-pending-update-create~1`
+
+Status: approved
+
+When the WorkloadScheduler receives the `WorkloadOperation` to update a workload
+and the workload is an inter-workload dependency of at least one other workload
+and the workload has fulfilled `DeleteCondition`s,
+then the WorkloadScheduler shall report the workload state `Pending(WaitingToStart)`.
+
+Rationale: This communicates to the user that the create operation of that workload is scheduled but not ready because the create operation of the update waits for inter-workload dependencies and the create operation shall only run after completing the delete operation of the update with the default update strategy `AT_MOST_ONCE`.
+
+Tags:
+- WorkloadScheduler
+
+Needs:
+- impl
+- utest
+- stest
+
+#### The agent shall not enqueue a delete only operation of an update
+`swdd~agent-shall-not-enqueue-update-delete-only-workload-operation~1`
+
+Status: approved
+
+The WorkloadScheduler shall not put the update delete only `WorkloadOperation` into a waiting queue.
+
+Rationale: The update delete only workload operation is internally created when the delete operation of a pending update is ready.
+
+Tags:
+- WorkloadScheduler
+
+Needs:
+- impl
+- utest
+
+#### A workload is ready to create when all of its inter-workload dependencies are fulfilled
+`swdd~workload-ready-to-create-on-fulfilled-dependencies~1`
+
+Status: approved
+
+When the workload state of all inter-workload dependencies of a workload fulfill their configured `AddCondition`, then the Workload is ready to create.
+
+Tags:
+- WorkloadScheduler
+- DependencyStateValidator
+
+Needs:
+- impl
+- utest
+
+#### An inter-workload dependency is ready to delete when all of its inter-workload dependencies are fulfilled
+`swdd~workload-ready-to-delete-on-fulfilled-dependencies~1`
+
+Status: approved
+
+When the workload states of all workloads fulfill the respective `DeleteCondition` of an inter-workload dependency, then the inter-workload dependency is ready to delete.
```

Comment:
Here I would keep "fulfill" because that is used in all other requirements as well. And here are more details about what "fulfill" means exactly. Alternatively, I can rename the whole "fulfill" terms into "meet". I am painless there...